test/socket.immediate.js

```diff
@@ -0,0 +1,39 @@
+var zmq = require('..')
+  , should = require('should')
+  , semver = require('semver');
+
+
+describe('socket.immediate', function() {
+	
+	it('should test ZMQ_IMMEDIATE socket option', function(done) {
+		
+		var msgsToSend = 5
+		  , receivedMsgs = 0;
+		
+		var push = zmq.socket('push');
+		push.setsockopt('immediate', 1);
+		
+		push.connect('tcp://127.0.0.1:5423');
+		
+		for(var i=0; i<msgsToSend; i++) {
+			push.send('message #'+i);
+		}
+		
+		var pull = zmq.socket('pull');
+		
+		var pull = zmq.socket('pull');
+		
+		pull.on('message', function(data) {
+			receivedMsgs += 1;
+		});
+		
+		pull.bind('tcp://127.0.0.1:5423', function(err){
+			if (err) throw err;
+		});
+		
+		setTimeout(function(){
+			receivedMsgs.should.equal(0);
+			done();
+		}, 1000);
```

Comment:
I've just updated the test unit, shall I create a new pull request ?
regarding of the file binding.cc, I'm not sure about what I did whether it correct or not, could you help please.